cli/mender-artifact/main.go

```diff
@@ -52,16 +52,7 @@ func applyCompressionInCommand(c *cli.Context) error {
 	// Let --compression argument work after command as well. Latest one
 	// applies.
 	if c.String("compression") != "" {
-		parent := c
-		// Find top level context, where the original --compression
-		// argument lives.
-		for {
-			if parent.Parent() == nil {
-				break
-			}
-			parent = parent.Parent()
-		}
-		parent.Set("compression", c.String("compression"))
+		c.GlobalSet("compression", c.String("compression"))
```

Comment:
haha, found the opportunity to fix this one up, huh

Reply:
I'm a man of my word!